internal/service/ec2/transitgateway_shared_owners.go

```diff
@@ -0,0 +1,172 @@
+		log.Printf("[WARN] Transit Gateway %s not found, removing from state", transitGatewayID)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("transit_gateway_id", transitGatewayID)
+
+	accountID := d.Get("account_id").(string)
+	d.Set("account_id", accountID)
+
+	if !d.IsNewResource() {
+		if transitGateway.Options != nil && transitGateway.Options.SharedOwners != nil {
+			AccountIDExist := false
+			for _, owner := range transitGateway.Options.SharedOwners {
+				if owner != nil && aws.StringValue(owner) == accountID {
+					AccountIDExist = true
+					break
+				}
+			}
+			if !AccountIDExist {
+				log.Printf("[WARN] In Transit Gateway Shared %s User %s not found, removing from state", d.Id(), accountID)
+				d.SetId("")
+				return nil
+			}
+		}
+	}
+	return nil
+}
+
+func ResourceTransitGatewaySharedOwnersDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	accountID := d.Get("account_id").(string)
+
+	modifyInput := &ec2.ModifyTransitGatewayInput{
+		TransitGatewayId: aws.String(transitGatewayID),
+		Options: &ec2.ModifyTransitGatewayOptions{
+			RemoveSharedOwners: []*string{aws.String(accountID)},
+		},
+	}
+
+	_, err := conn.ModifyTransitGateway(modifyInput)
+	if err != nil {
+		log.Printf("[WARN] Error removing shared owner %s from Transit Gateway %s: %s", accountID, transitGatewayID, err)
+	}
+
+	d.SetId("")
+
+	log.Printf("[INFO] Removed shared owner %s from Transit Gateway %s", accountID, transitGatewayID)
+
+	return nil
+}
+
+const (
+	transitGatewaySharedOwnersIDSeparator = "/"
+)
+
+func resourceTransitGatewaySharedOwnersImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), transitGatewaySharedOwnersIDSeparator)
+
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		return nil, fmt.Errorf("unexpected format for ID (%[1]s), expected TRANSIT-GATEWAY-ID%[2]sACCOUNT-ID", d.Id(), transitGatewaySharedOwnersIDSeparator)
+	}
+
+	transitGatewayID := parts[0]
+	accountID := parts[1]
+
+	d.SetId(TransitGatewaySharedOwnersCreateResourceID(transitGatewayID, accountID))
+	d.Set("transit_gateway_id", transitGatewayID)
+	d.Set("account_id", accountID)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func TransitGatewaySharedOwnersCreateResourceID(transitGatewayID, accountID string) string {
+	return fmt.Sprintf("%s%s%s", transitGatewayID, transitGatewaySharedOwnersIDSeparator, accountID)
+}
+
+func TransitGatewaySharedOwnersParseResourceID(id string) (string, string, string, string, string, error) {
+	parts := strings.Split(id, transitGatewaySharedOwnersIDSeparator)
+
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		return "", "", "", "", "", fmt.Errorf("unexpected format for ID (%[1]s), expected TRANSIT-GATEWAY-ID%[2]sACCOUNT-ID", id, transitGatewaySharedOwnersIDSeparator)
```

Comment:
```suggestion
		return "", "", "", "", "", fmt.Errorf("unexpected format for ID (%s), expected TRANSIT-GATEWAY-ID%sACCOUNT-ID", id, transitGatewaySharedOwnersIDSeparator)
```

По-моему, тут можно без индексов